### PR TITLE
Issue/101

### PR DESCRIPTION
**Pull request by shmuga**
Fixing #101 #102

### CODE REVIEW

File: src/Request/RequestErrors.js [hunk 1]
```diff
@@ -19,12 +19,13 @@ Object.assign(RequestValidationError, createErrorsList({
 
 // Runtime errors
 export const RequestRuntimeError = createErrorClass(
-  'RequestRuntimeError',
+  'ParserRuntimeError',
```
Comment (dchertousov) [inline src/Request/RequestErrors.js]: Why do we have this name change here?

Comment (dchertousov) [inline src/Request/RequestErrors.js]: As you still keep exporting `RequestRuntimeError`...

File: src/Request/uapi-parser.js [hunk 2]
```diff
@@ -258,6 +284,7 @@ Parser.prototype.parse = function (xml) {
       if (self.debug > 2) {
         console.log('Error during automatic resolving version');
       }
+      throw new RequestRuntimeError.VersionParsingError(obj, e);
```
Comment (dchertousov) [inline src/Request/uapi-parser.js]: It's not yet necessary, as we still have fallback version. But I'd like to keep it.
If we decide to keep it, then debug is not needed.

File: test/parser.test.js [hunk 3]
```diff
@@ -1,15 +1,26 @@
 const assert = require('assert');
+const fs = require('fs');
+
 const errors = require('../src').errors;
 const UAPIParser = require('../src/Request/uapi-parser');
 
-const parser = new UAPIParser('v_36_0', {});
+const xmlFolder = __dirname + '/FakeResponses';
```
Comment (dchertousov) [inline test/parser.test.js]: Please, use `path.join()` instead.

File: src/Request/uapi-parser.js (hunk 2 above)
Comment (shmuga) [inline src/Request/uapi-parser.js]: Let's keep this error throw as a part of fallback version deprecation.

**Comment by shmuga** [comment]
@dchertousov recheck and approve pls

File: test/parser.test.js [hunk 4]
```diff
@@ -1,10 +1,11 @@
 const assert = require('assert');
 const fs = require('fs');
+const path = require('path');
 
 const errors = require('../src').errors;
 const UAPIParser = require('../src/Request/uapi-parser');
 
-const xmlFolder = __dirname + '/FakeResponses';
+const xmlFolder = path.join(__dirname, '/FakeResponses');
```
Comment (dchertousov) [inline test/parser.test.js]: If you use join, then '/' is not needed.

File: test/parser.test.js [hunk 5]
```diff
@@ -18,7 +19,7 @@ describe('parser tests', () => {
 
   it('should detect error version automaticly', () => {
     const parser = new UAPIParser('someroot', 'v36_0');
-    const xml = fs.readFileSync(`${xmlFolder}/Other/UnableToFareQuoteError.xml`);
+    const xml = fs.readFileSync(path.join(xmlFolder, '/Other/UnableToFareQuoteError.xml'));
```
Comment (dchertousov) [inline test/parser.test.js]: Same goes here.

File: src/Request/uapi-parser.js (hunk 2 above)
Comment (dchertousov) [inline src/Request/uapi-parser.js]: I'd like to keep it too!
I meant removing these lines:
```javascript
       if (self.debug > 2) {
         console.log('Error during automatic resolving version');
       }
```

As we don't need logging if we are throwing an error

**Comment by dchertousov** [comment]
Waiting for changes discussed in IM

**Comment by shmuga** [comment]
@dchertousov done!

**Pull request merged**